### PR TITLE
(PCP-785) Report task download failure for 400+ HTTP statuses

### PR DESCRIPTION
**Pull request by ekinanp**
Previously, when the task download returned a 400+ response, the
response itself was downloaded as the task file. This resulted in
a SHA-version conflict PXP-error being returned, instead of a
download error. This has now been fixed.

### CODE REVIEW

**Comment by ekinanp** [comment]
This should be merged after https://github.com/puppetlabs/leatherman/pull/259.

File: lib/src/modules/task.cc [hunk 1]
```diff
@@ -260,7 +260,13 @@ static std::tuple<bool, std::string> downloadTaskFile(const std::vector<std::str
         // timeout from connection after one minute, can configure
         req.connection_timeout(60000);
         try {
-            client.download_file(req, file_path.string(), NIX_TASK_FILE_PERMS);
+            auto resp = client.download_file(req, file_path.string(), NIX_TASK_FILE_PERMS);
```
Comment (MikaelSmith) [inline lib/src/modules/task.cc]: Needs to be reworked for the new function signature.

**Comment by MikaelSmith** [comment]
https://github.com/puppetlabs/leatherman/pull/260 preps release of Leatherman.

**Comment by ekinanp** [comment]
@MikaelSmith updated with the new function signature in leatherman

**Comment by MikaelSmith** [comment]
Leatherman should be building packages now. When they're ready at https://github.com/puppetlabs/leatherman/releases/tag/1.2.1, update the Travis and AppVeyor config to use 1.2.1.

**Comment by ekinanp** [comment]
@MikaelSmith CI passed

**Pull request merged**